[MISC] Clarify participant_id in participants.tsv file if it exists (#738)
bids-standard/bids-specification: src/06-longitudinal-and-multi-site-studies.md

# Longitudinal and multi-site studies

Multiple sessions (visits) are encoded by adding an extra layer of directories

> and file names in the form of `ses-<label>`. Session label MUST consist

Reviewer: ```suggestion
and file names in the form of `ses-<label>`. Session labels MUST consist
```

> and file names in the form of `ses-<label>`. Session labels MUST consist
> only of alphanumeric characters `[a-zA-Z0-9]` and SHOULD be consistent across
> subjects. If numbers are used in session labels we RECOMMEND using zero padding

Reviewer: ```suggestion
and file names in the form of `ses-<label>`.
Session labels MUST consist only of alphanumeric characters `[a-zA-Z0-9]`
and SHOULD be consistent across subjects.
If numbers are used in session labels we RECOMMEND using zero padding
```

(for example `ses-01`, `ses-11` instead of `ses-1`, `ses-11`). This makes
results of alphabetical sorting more intuitive. Acquisition time of session can
be defined in the [sessions file](#sessions-file).

The extra session layer (at least one `/ses-<label>` subfolder) SHOULD
be added for all subjects if at least one subject in the dataset has more than
one session. If a `/ses-<label>` subfolder is included as part of the

> directory hierarchy, then the same `ses-<label>` tag MUST also be

Reviewer: > Skipping the session layer for only some subjects in the dataset is
not allowed

I actually think that the original text here meant to say that:

> The extra session layer (at least one `/ses-<label>` subfolder) MUST
be added for all subjects if at least one subject in the dataset has more than
one session.

instead of the SHOULD you used here.

opinions @Remi-Gau @tsalo ?

Reviewer: on the other hand, that doesn't really make sense, because there are experiments where some subjects perform an extra session :thinking:

Author: I think skipping the `ses-<X>` in some subjects is allowed if say not all subjects have the same sessions. Rn this results in a warning in validator, which basically all my data has this warning :p.

```The extra session layer (at least one /ses-<label> subfolder) MUST
be added for all subjects if at least one subject in the dataset has more than
one session.```

seems reasonable to me... The problem is how does one validate this? If one has multiple "sessions" per subject, then that implies there are potentially different electrode/channels across those sessions, so the user would want to store it as such. Does this basically mean if one has:

```
sub-01/
   sub-01_ses-01_task-blah_ieeg.vhdr
   ...
```
it's allowed, but if you have:

```
sub-01/
   sub-01_ses-01_task-blah_ieeg.vhdr
   sub-01_ses-02_task-blah_ieeg.vhdr
   ...
```
it's not?

Reviewer: > I think skipping the ses-<X> in some subjects is allowed if say not all subjects have the same sessions. Rn this results in a warning in validator, which basically all my data has this warning

yes, I experienced that warning multiple times myself :roll_eyes: I think this case is more common than thought when this warning was originally implemented. That is, more often it's a choice, rather than an overlooked error.

re: your example --> yes :thinking: but I think your current changes (apart from my comments below) are actually fine and clarify the situation without changing the state of anything (which is what we should aim for), so I'd be fine with leaving the SHOULD, not interpreting it as a MUST post-hoc.

Reviewer 2: Wait wasn't the original intent of theb MUST to avoid things like:

```
├── sub-1
│   └── eeg
└── sub-2
    ├── ses-1
    │   └── eeg
    └── ses-2
        └── fmri
```

Or maybe I am missing something?

Reviewer: yes, that's how I interpret it as well. What's annoying is that first the text says "should" (not MUST), and then the follow up sentence is:

> Skipping the session layer for only some subjects in the dataset is not allowed.

so basically the "should" is actually a MUST.

However:

1. I am not sure why this rule makes sense
2. I am pretty sure that I have seen datasets not conforming to this rule
3. There is no validator support for this rule as far as I know

:roll_eyes: perhaps we can discuss this tonight in the maintainers meeting

Reviewer 2: Yeah the inconsistency should be fixed or clarified.

I mean I find the folder structure I described above "ugly", but thank goodness we don't just rely on my sense of aesthetics.

:wink:

> directory hierarchy, then the same `ses-<label>` tag MUST also be

Reviewer 3: ```suggestion
directory hierarchy, then the same [`ses-<label>`](../99-appendices/09-entities.md#ses) key/value pair MUST also be
```

> one session. If a `/ses-<label>` subfolder is included as part of the
> directory hierarchy, then the same [`ses-<label>`](./99-appendices/09-entities.md#ses) key/value pair MUST also be

Reviewer: ```suggestion
one session.
If a `/ses-<label>` subfolder is included as part of the directory hierarchy,
then the same [`ses-<label>`](./99-appendices/09-entities.md#ses)
key/value pair MUST also be
```

Optional: Yes

In case of multiple sessions there is an option of adding additional

> ``sessions.tsv`` files describing variables changing between sessions. In such

Reviewer: ```suggestion
`sessions.tsv` files describing variables changing between sessions. In such
```

> case one file per participant should be added. These files MUST include

Reviewer: ```suggestion
case one file per participant SHOULD be added. These files MUST include a
```

> compulsory `session_id` column and describe each session by one and only one

Reviewer: ```suggestion
`session_id` column and describe each session by one and only one row.
```

> row. Column names in per ``sessions.tsv`` files MUST be different from group
> level participant key column names in the ``participants.tsv`` file.

Reviewer: ```suggestion
Column names in `sessions.tsv` files MUST be different from group
level participant key column names in the `participants.tsv` file.
```

> `session_id` column and describe each session by one and only one row.
> Column names in `sessions.tsv` files MUST be different from group
> level participant key column names in the `participants.tsv` file.

Reviewer: following from "participants.tsv":

> that MUST consist of `sub-<label>` identifying each row for a
participant, followed by a list of optional columns describing participants.

```suggestion
`session_id` column that MUST consist of `ses-<label>`, and that
MUST describe each session by one and only one row.
Column names in `sessions.tsv` files MUST be different from group
level participant key column names in the `participants.tsv` file.
```

Reviewer: could you please also add a link back to "participants file" in the last two words of this sentence?

Author: Done

> level participant key column names in the [`participants.tsv` file](./03-modality-agnostic-files.md).

Reviewer: ```suggestion
level participant key column names in the
[`participants.tsv` file](./03-modality-agnostic-files.md#participants-file).
```

Optional: Yes

In case of multiple sessions there is an option of adding additional

> `sessions.tsv` files describing variables changing between sessions. In such
> case one file per participant SHOULD be added. These files MUST include a
> `session_id` column and describe each session by one and only one row.
> Column names in `sessions.tsv` files MUST be different from group
> level participant key column names in the
> [`participants.tsv` file](./03-modality-agnostic-files.md#participants-file).

Reviewer: ```suggestion
`sessions.tsv` files describing variables changing between sessions.
In such case one file per participant SHOULD be added.
These files MUST include a `session_id` column and describe each session by one and only one row.
Column names in `sessions.tsv` files MUST be different from group level participant key column names in the
[`participants.tsv` file](./03-modality-agnostic-files.md#participants-file).
```
